src/pages/archivePage/ArchivePage.jsx

```diff
@@ -1,9 +1,50 @@
+import axios from "axios";
 import { Archive } from "components/index";
+import { useNotes } from "context";
+import { useEffect } from "react";
+import "./archivePage.css";
 
 function ArchivePage() {
+  const encodedToken = localStorage.getItem("token");
+  const {
+    noteState: { archivedNotes },
+    noteDispatch,
+  } = useNotes();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = axios.get("/api/archives", {
+          headers: {
+            authorization: encodedToken,
+          },
+        });
+        if (response.status === 200) {
+          noteDispatch({
+            type: "SET_ARCHIVE",
+            payload: (await response).data.archives,
+          });
+        }
+      } catch (err) {
+        console.log(err);
+      }
+    })();
+  }, [encodedToken, noteDispatch]);
```

Comment:
Here It is better give an empty dependency array for the use effect because you need to fetch the archived notes data only on the very first render of the page, that is when you log in. Once you are logged in you need not do this every time when the noteDispatch is being called because you are updating your reducer state in the reducer function itself.

Reply:
I tried it but vsCode gives a warning to put encodedToken and noteDispatch in the dependency array. So included them to get rid of warning.